### ISSUE TITLE
Fre:ac CMD doesn't understand arguments when there is Kanji (Japanese characters) in file path

### ISSUE THREAD

**Issue by Arno500** [enzo1982/freac#41]
**Describe the bug**
When trying to convert a file with Kanji in it (on Windows), Fre:ac only displays basic help, like if we used an invalid argument.
The problem doesn't exists in the main Fre:ac GUI application.

**To Reproduce**
In my case, the file was :
C:\Users\Arno DUBOIS\Desktop\Stack\Music\君の名は。 - RADWIMPS (24.08.2016) [FLAC]\01 夢灯籠.flac  
1. Run fre:ac with a file containing Japanese as input (in the file name or path).

**Expected behavior**
Fre:ac should run the conversion.

**System (please complete the following information)**
 - OS: Windows 10 May Update (1903)
 - CPU: Intel Core i7 7700HQ
 - RAM: 16 GB

**Comment by Arno500** [comment]
Oh, and I also tried your latest build from the 20190610 (from the issue #39), but it was before your file encoding modifications to your smooth library. And you don't provide a lot of informations to build Fre:ac on windows, so I kind of dropped it.

**Comment by enzo1982** [comment]
Well, it works for me, see this screenshot made with freaccmd 20190423:

![2019-07-07 233733](https://user-images.githubusercontent.com/7796713/60774172-a87fb080-a110-11e9-82c0-d62c10cc1767.png)

This is using the new Windows Terminal preview app which is the first to really support Unicode. It works with the regular console too, but you will not be able to input Japanese characters there. You can use tab completion, which will print placeholders for the CJK characters but still pass the correct file name to fre:ac.

fre:ac still prints ??? when processing the file, but processes it correctly. I'll see if I can fix the output to print in Unicode mode.

So if this doesn't work for you, something else must be wrong on your end.

**Comment by Arno500** [comment]
In fact, I was calling it from a PowerShell script. It loops over every
file in my folder and convert them to Opus. I tried manually converting the
files (still from the PowerShell console) to no avail. Also tried the new
Windows Terminal.
I also tried to set the console to UTF8 by hands.
Note that the file are perfectly processed by other utilities.
I'm starting to think there is a compatibility problem with PowerShell.

Le dim. 7 juil. 2019 à 23:47, Robert Kausch <notifications@github.com> a
écrit :

**Comment by enzo1982** [comment]
I just pushed a [commit](https://github.com/enzo1982/smooth/commit/51ed23) with a few fixes. Unicode I/O on the Windows console should work as expected now:

![screenshot](https://user-images.githubusercontent.com/7796713/60837801-6a929300-a1c9-11e9-9f09-6083f1498c37.png)

I also prepared a build for you to test: [freac-1.1-alpha-20190708-windows-x64.zip](https://github.com/enzo1982/freac/files/3369995/freac-1.1-alpha-20190708-windows-x64.zip)

fre:ac actually was not getting passed Unicode file names on the command line, due to using the non-Unicode command line argument passed to WinMain. Unicode characters were passed as `?` causing them to be interpreted as wildcards. That's why it still worked for me when I first tried. fre:ac just coincidentally found the correct file using the wildcard pattern.

The fix is to use the `GetCommandLineW()` function instead to get a Unicode version of the command line.

The console output issue was due to using the default ANSI codepage and was easily fixed by switching to UTF-8 with `SetConsoleOutputCP()`.
